<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Generating Various Numerical Representation Schemes for Protein Sequences • protr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Generating Various Numerical Representation Schemes for Protein Sequences">
<meta property="og:description" content="Comprehensive toolkit for generating various numerical
    features of protein sequences described in Xiao et al. (2015)
    &lt;DOI:10.1093/bioinformatics/btv042&gt;. For full functionality,
    the software 'ncbi-blast+' is needed, see
    &lt;https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download&gt;
    for more information.">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/protr.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/road2stat/protr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="contents col-md-9">
<div id="protr-" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#protr-" class="anchor"></a>protr <a href="https://nanx.me/protr/"><img src="https://i.imgur.com/D0nDOWs.png" align="right" alt="logo" height="180" width="180"></a>
</h1></div>

<p>Comprehensive toolkit for generating various numerical features of protein sequences described in Xiao et al. (2015) &lt;<a href="https://academic.oup.com/bioinformatics/article-lookup/doi/10.1093/bioinformatics/btv042">DOI:10.1093/bioinformatics/btv042</a>&gt; (<a href="https://nanx.me/papers/protr.pdf">PDF</a>).</p>
<div id="paper-citation" class="section level2">
<h2 class="hasAnchor">
<a href="#paper-citation" class="anchor"></a>Paper Citation</h2>
<p>Formatted citation:</p>
<p>Nan Xiao, Dong-Sheng Cao, Min-Feng Zhu, and Qing-Song Xu. (2015). protr/ProtrWeb: R package and web server for generating various numerical representation schemes of protein sequences. <em>Bioinformatics</em> 31 (11), 1857-1859.</p>
<p>BibTeX entry:</p>
<pre><code>@article{Xiao2015,
  author = {Xiao, Nan and Cao, Dong-Sheng and Zhu, Min-Feng and Xu, Qing-Song.},
  title = {{protr/ProtrWeb: R package and web server for generating various numerical representation schemes of protein sequences}},
  journal = {Bioinformatics},
  year = {2015},
  volume = {31},
  number = {11},
  pages = {1857--1859},
  doi = {10.1093/bioinformatics/btv042},
  issn = {1367-4803},
  url = {http://bioinformatics.oxfordjournals.org/content/31/11/1857}
}</code></pre>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>To install <code>protr</code> from CRAN:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/install.packages">install.packages</a></span>(<span class="st">"protr"</span>)</a></code></pre></div>
<p>Or try the latest version on GitHub:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="co"># install.packages("devtools")</span></a>
<a class="sourceLine" id="cb3-2" title="2">devtools<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/devtools/topics/reexports">install_github</a></span>(<span class="st">"road2stat/protr"</span>)</a></code></pre></div>
<p><a href="https://nanx.me/protr/articles/protr.html">Browse the package vignette</a> for a quick-start.</p>
</div>
<div id="shiny-web-application" class="section level2">
<h2 class="hasAnchor">
<a href="#shiny-web-application" class="anchor"></a>Shiny Web Application</h2>
<p>ProtrWeb, the Shiny web application built on protr, can be accessed from <a href="http://protr.org">http://protr.org</a>.</p>
<p>ProtrWeb is a user-friendly web application for computing the protein sequence descriptors (features) presented in the protr package.</p>
</div>
<div id="descriptors-list" class="section level2">
<h2 class="hasAnchor">
<a href="#descriptors-list" class="anchor"></a>Descriptors List</h2>
<div id="commonly-used-descriptors" class="section level3">
<h3 class="hasAnchor">
<a href="#commonly-used-descriptors" class="anchor"></a>Commonly used descriptors</h3>
<ul>
<li>Amino acid composition descriptors
<ul>
<li>Amino acid composition</li>
<li>Dipeptide composition</li>
<li>Tripeptide composition</li>
</ul>
</li>
<li>Autocorrelation descriptors
<ul>
<li>Normalized Moreau-Broto autocorrelation</li>
<li>Moran autocorrelation</li>
<li>Geary autocorrelation</li>
</ul>
</li>
<li>CTD descriptors
<ul>
<li>Composition</li>
<li>Transition</li>
<li>Distribution</li>
</ul>
</li>
<li><p>Conjoint Triad descriptors</p></li>
<li>Quasi-sequence-order descriptors
<ul>
<li>Sequence-order-coupling number</li>
<li>Quasi-sequence-order descriptors</li>
</ul>
</li>
<li>Pseudo amino acid composition (PseAAC)
<ul>
<li>Pseudo amino acid composition</li>
<li>Amphiphilic pseudo amino acid composition</li>
</ul>
</li>
<li>Profile-based descriptors
<ul>
<li>Profile-based descriptors derived by PSSM (Position-Specific Scoring Matrix)</li>
</ul>
</li>
</ul>
</div>
<div id="proteochemometric-pcm-modeling-descriptors" class="section level3">
<h3 class="hasAnchor">
<a href="#proteochemometric-pcm-modeling-descriptors" class="anchor"></a>Proteochemometric (PCM) modeling descriptors</h3>
<ul>
<li>Scales-based descriptors derived by principal components analysis
<ul>
<li>Scales-based descriptors derived by amino acid properties (AAindex)</li>
<li>Scales-based descriptors derived by 20+ classes of 2D and 3D molecular descriptors (Topological, WHIM, VHSE, etc.)</li>
<li>Scales-based descriptors derived by factor analysis</li>
<li>Scales-based descriptors derived by multidimensional scaling</li>
<li>BLOSUM and PAM matrix-derived descriptors</li>
</ul>
</li>
</ul>
</div>
<div id="similarity-computation" class="section level3">
<h3 class="hasAnchor">
<a href="#similarity-computation" class="anchor"></a>Similarity Computation</h3>
<p>Local and global pairwise sequence alignment for protein sequences:</p>
<ul>
<li>Between two protein sequences</li>
<li>Parallelized pairwise similarity calculation with a list of protein sequences</li>
</ul>
<p>GO semantic similarity measures:</p>
<ul>
<li>Between two groups of GO terms / two Entrez Gene IDs</li>
<li>Parallelized pairwise similarity calculation with a list of GO terms / Entrez Gene IDs</li>
</ul>
</div>
<div id="miscellaneous-tools-and-datasets" class="section level3">
<h3 class="hasAnchor">
<a href="#miscellaneous-tools-and-datasets" class="anchor"></a>Miscellaneous tools and datasets</h3>
<ul>
<li>Retrieve protein sequences from UniProt</li>
<li>Read protein sequences in FASTA format</li>
<li>Read protein sequences in PDB format</li>
<li>Sanity check of the amino acid types appeared in the protein sequences</li>
<li>Protein sequence segmentation</li>
<li>Auto cross covariance (ACC) for generating scales-based descriptors of the same length</li>
<li>20+ pre-computed 2D and 3D descriptor sets for the 20 amino acids to use with the scales-based descriptors</li>
<li>BLOSUM and PAM matrices for the 20 amino acids</li>
<li>Meta information of the 20 amino acids</li>
</ul>
</div>
</div>
<div id="links" class="section level2">
<h2 class="hasAnchor">
<a href="#links" class="anchor"></a>Links</h2>
<ul>
<li>Website: <a href="https://nanx.me/protr/" class="uri">https://nanx.me/protr/</a>
</li>
<li>CRAN: <a href="https://cran.r-project.org/package=protr" class="uri">https://cran.r-project.org/package=protr</a>
</li>
<li>GitHub: <a href="https://github.com/road2stat/protr" class="uri">https://github.com/road2stat/protr</a>
</li>
</ul>
</div>
<div id="contribute" class="section level2">
<h2 class="hasAnchor">
<a href="#contribute" class="anchor"></a>Contribute</h2>
<p>To contribute to this project, please take a look at the <a href="CONTRIBUTING.html">Contributing Guidelines</a> first. Please note that this project is released with a <a href="CONDUCT.html">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Download from CRAN at <br><a href="https://cloud.r-project.org/package=protr">https://​cloud.r-project.org/​package=protr</a>
</li>
<li>Browse source code at <br><a href="https://github.com/road2stat/protr">https://​github.com/​road2stat/​protr</a>
</li>
<li>Report a bug at <br><a href="https://github.com/road2stat/protr/issues">https://​github.com/​road2stat/​protr/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing protr</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Nan Xiao <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-0250-5673" target="orcid.widget"><img src="https://members.orcid.org/sites/default/files/vector_iD_icon.svg" class="orcid" alt="ORCID" height="16"></a> </li>
<li>Qing-Song Xu <br><small class="roles"> Author </small>  </li>
<li>Dong-Sheng Cao <br><small class="roles"> Author </small>  </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/road2stat/protr"><img src="https://travis-ci.org/road2stat/protr.svg?branch=master" alt="Build Status"></a></li>
<li><a href="https://ci.appveyor.com/project/road2stat/protr"><img src="https://ci.appveyor.com/api/projects/status/github/road2stat/protr?branch=master&amp;svg=true" alt="AppVeyor Build Status"></a></li>
<li><a href="https://cran.r-project.org/package=protr"><img src="https://www.r-pkg.org/badges/version/protr" alt="CRAN Version"></a></li>
<li><a href="https://cranlogs.r-pkg.org/badges/protr"><img src="https://cranlogs.r-pkg.org/badges/protr" alt="Downloads from the RStudio CRAN mirror"></a></li>
</ul>
</div>
</div>
</div>

      <footer><div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
