<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Scales-Based Descriptors derived by Factor Analysis — extractFAScales • protr</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Scales-Based Descriptors derived by Factor Analysis — extractFAScales" />

<meta property="og:description" content="This function calculates scales-based descriptors
derived by Factor Analysis (FA).
Users can provide customized amino acid property matrices." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/road2stat/protr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Scales-Based Descriptors derived by Factor Analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/road2stat/protr/blob/master/R/pcm-04-extractFAScales.R'><code>R/pcm-04-extractFAScales.R</code></a></small>
    <div class="hidden name"><code>extractFAScales.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>This function calculates scales-based descriptors
derived by Factor Analysis (FA).
Users can provide customized amino acid property matrices.</p>
    
    </div>

    <pre class="usage"><span class='fu'>extractFAScales</span>(<span class='no'>x</span>, <span class='no'>propmat</span>, <span class='no'>factors</span>, <span class='kw'>scores</span> <span class='kw'>=</span> <span class='st'>"regression"</span>, <span class='no'>lag</span>,
  <span class='kw'>scale</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A character vector, as the input protein sequence.</p></td>
    </tr>
    <tr>
      <th>propmat</th>
      <td><p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents one property.
Note that the one-letter row names must be provided for we need them to seek
the properties for each AA type.</p></td>
    </tr>
    <tr>
      <th>factors</th>
      <td><p>Integer. The number of factors to be fitted.
Must be no greater than the number of AA properties provided.</p></td>
    </tr>
    <tr>
      <th>scores</th>
      <td><p>Type of scores to produce. The default is <code>"regression"</code>,
which gives Thompson's scores, <code>"Bartlett"</code> given Bartlett's weighted
least-squares scores.</p></td>
    </tr>
    <tr>
      <th>lag</th>
      <td><p>The lag parameter. Must be less than the amino acids number
in the protein sequence.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing Factor Analysis? Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>Logical. Whether we print the SS loadings,
proportion of variance and the cumulative proportion of
the selected factors or not. Default is <code>TRUE</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (factors) selected.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Atchley, W. R., Zhao, J., Fernandes, A. D., &amp; Druke, T. (2005).
Solving the protein sequence metric problem.
Proceedings of the National Academy of Sciences of the United States of America,
102(18), 6395-6400.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>x</span> <span class='kw'>=</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/system.file'>system.file</a></span>(<span class='st'>"protseq/P00750.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>AATopo</span>)
<span class='no'>tprops</span> <span class='kw'>=</span> <span class='no'>AATopo</span>[, <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>37</span>:<span class='fl'>41</span>, <span class='fl'>43</span>:<span class='fl'>47</span>)]  <span class='co'># select a set of topological descriptors</span>
<span class='no'>fa</span> <span class='kw'>=</span> <span class='fu'>extractFAScales</span>(
  <span class='no'>x</span>, <span class='kw'>propmat</span> <span class='kw'>=</span> <span class='no'>tprops</span>, <span class='kw'>factors</span> <span class='kw'>=</span> <span class='fl'>5</span>, <span class='kw'>lag</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; Summary of the factor analysis result:
#&gt; 
#&gt; Call:
#&gt; factanal(x = propmat, factors = factors, scores = scores)
#&gt; 
#&gt; Uniquenesses:
#&gt; WhetZ Whetm Whetv Whete Whetp JhetZ Jhetm Jhetv Jhete Jhetp 
#&gt; 0.005 0.005 0.005 0.005 0.005 0.005 0.005 0.005 0.005 0.005 
#&gt; 
#&gt; Loadings:
#&gt;       Factor1 Factor2 Factor3 Factor4 Factor5
#&gt; WhetZ  0.982  -0.166                         
#&gt; Whetm  0.982  -0.166                         
#&gt; Whetv  0.989  -0.133                         
#&gt; Whete  0.982  -0.159                         
#&gt; Whetp  0.989  -0.131                         
#&gt; JhetZ -0.399   0.881   0.162   0.192         
#&gt; Jhetm -0.399   0.881   0.162   0.192         
#&gt; Jhetv          0.985          -0.131         
#&gt; Jhete -0.378   0.872   0.300                 
#&gt; Jhetp          0.970  -0.210                 
#&gt; 
#&gt;                Factor1 Factor2 Factor3 Factor4 Factor5
#&gt; SS loadings      5.313   4.340   0.212   0.107   0.003
#&gt; Proportion Var   0.531   0.434   0.021   0.011   0.000
#&gt; Cumulative Var   0.531   0.965   0.986   0.997   0.997
#&gt; 
#&gt; Test of the hypothesis that 5 factors are sufficient.
#&gt; The chi square statistic is 472.2 on 5 degrees of freedom.
#&gt; The p-value is 7.96e-100 </div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#references">References</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Nan Xiao &lt;<a href='https://nanx.me'>https://nanx.me</a>&gt;</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

