<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Scales-Based Descriptors with 20+ classes of Molecular Descriptors — extractDescScales • protr</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png" />
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png" />
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png" />
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png" />

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Scales-Based Descriptors with 20+ classes of Molecular Descriptors — extractDescScales" />
<meta property="og:description" content="This function calculates the scales-based descriptors with
molecular descriptors sets calculated by Dragon, Discovery Studio and MOE.
Users can specify which molecular descriptors to select from one of these
deseriptor sets by specify the numerical or character index of the
molecular descriptors in the descriptor set." />
<meta property="og:image" content="/logo.png" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.6-2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nanxstats/protr/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Scales-Based Descriptors with 20+ classes of Molecular Descriptors</h1>
    <small class="dont-index">Source: <a href='https://github.com/nanxstats/protr/blob/master/R/pcm-02-extractDescScales.R'><code>R/pcm-02-extractDescScales.R</code></a></small>
    <div class="hidden name"><code>extractDescScales.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function calculates the scales-based descriptors with
molecular descriptors sets calculated by Dragon, Discovery Studio and MOE.
Users can specify which molecular descriptors to select from one of these
deseriptor sets by specify the numerical or character index of the
molecular descriptors in the descriptor set.</p>
    </div>

    <pre class="usage"><span class='fu'>extractDescScales</span><span class='op'>(</span>
  <span class='va'>x</span>,
  <span class='va'>propmat</span>,
  index <span class='op'>=</span> <span class='cn'>NULL</span>,
  <span class='va'>pc</span>,
  <span class='va'>lag</span>,
  scale <span class='op'>=</span> <span class='cn'>TRUE</span>,
  silent <span class='op'>=</span> <span class='cn'>TRUE</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A character vector, as the input protein sequence.</p></td>
    </tr>
    <tr>
      <th>propmat</th>
      <td><p>The matrix containing the descriptor set for the amino acids,
which can be chosen from
<code>AAMOE2D</code>, <code>AAMOE3D</code>, <code>AACPSA</code>, <code>AADescAll</code>,
<code>AA2DACOR</code>, <code>AA3DMoRSE</code>, <code>AAACF</code>, <code>AABurden</code>,
<code>AAConn</code>, <code>AAConst</code>, <code>AAEdgeAdj</code>, <code>AAEigIdx</code>,
<code>AAFGC</code>, <code>AAGeom</code>, <code>AAGETAWAY</code>, <code>AAInfo</code>,
<code>AAMolProp</code>, <code>AARandic</code>, <code>AARDF</code>, <code>AATopo</code>,
<code>AATopoChg</code>, <code>AAWalk</code>, and <code>AAWHIM</code>.</p></td>
    </tr>
    <tr>
      <th>index</th>
      <td><p>Integer vector or character vector. Specify which
molecular descriptors to select from one of these deseriptor
sets by specify the numerical or character index of the molecular
descriptors in the descriptor set.
Default is <code>NULL</code>, which means selecting all the molecular
descriptors in this descriptor set.</p></td>
    </tr>
    <tr>
      <th>pc</th>
      <td><p>Integer. The maximum dimension of the space which the data
are to be represented in. Must be no greater than the number of
amino acid properties provided.</p></td>
    </tr>
    <tr>
      <th>lag</th>
      <td><p>The lag parameter. Must be less than the amino acids.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before doing MDS? Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>Logical. Whether we print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales selected.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Nan Xiao &lt;<a href='https://nanx.me'>https://nanx.me</a>&gt;</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>x</span> <span class='op'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"protseq/P00750.fasta"</span>, package <span class='op'>=</span> <span class='st'>"protr"</span><span class='op'>)</span><span class='op'>)</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span>
<span class='va'>descscales</span> <span class='op'>&lt;-</span> <span class='fu'>extractDescScales</span><span class='op'>(</span>
  <span class='va'>x</span>,
  propmat <span class='op'>=</span> <span class='st'>"AATopo"</span>, index <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>37</span><span class='op'>:</span><span class='fl'>41</span>, <span class='fl'>43</span><span class='op'>:</span><span class='fl'>47</span><span class='op'>)</span>,
  pc <span class='op'>=</span> <span class='fl'>5</span>, lag <span class='op'>=</span> <span class='fl'>7</span>, silent <span class='op'>=</span> <span class='cn'>FALSE</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; Summary of the first 5 principal components: 
#&gt;                             PC1      PC2       PC3       PC4        PC5
#&gt; Standard deviation     2.581537 1.754133 0.4621854 0.1918666 0.08972087
#&gt; Proportion of Variance 0.666430 0.307700 0.0213600 0.0036800 0.00080000
#&gt; Cumulative Proportion  0.666430 0.974130 0.9954900 0.9991700 0.99998000</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


