<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Scales-Based Descriptors derived by Principal Components Analysis
(with Gap Support) — extractScalesGap • protr</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Scales-Based Descriptors derived by Principal Components Analysis
(with Gap Support) — extractScalesGap" />
<meta property="og:description" content="This function calculates scales-based descriptors
derived by Principal Components Analysis (PCA), with gap support.
Users can provide customized amino acid property matrices.
This function implements the core computation procedure needed for
the scales-based descriptors derived by AA-Properties (AAindex)
and scales-based descriptors derived by 20+ classes of 2D and 3D
molecular descriptors (Topological, WHIM, VHSE, etc.) in the protr package." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.6-2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nanxstats/protr/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Scales-Based Descriptors derived by Principal Components Analysis
(with Gap Support)</h1>
    <small class="dont-index">Source: <a href='https://github.com/nanxstats/protr/blob/master/R/pcm-01-extractScalesGap.R'><code>R/pcm-01-extractScalesGap.R</code></a></small>
    <div class="hidden name"><code>extractScalesGap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function calculates scales-based descriptors
derived by Principal Components Analysis (PCA), with gap support.
Users can provide customized amino acid property matrices.
This function implements the core computation procedure needed for
the scales-based descriptors derived by AA-Properties (AAindex)
and scales-based descriptors derived by 20+ classes of 2D and 3D
molecular descriptors (Topological, WHIM, VHSE, etc.) in the protr package.</p>
    </div>

    <pre class="usage"><span class='fu'>extractScalesGap</span>(<span class='no'>x</span>, <span class='no'>propmat</span>, <span class='no'>pc</span>, <span class='no'>lag</span>, <span class='kw'>scale</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>x</th>
      <td><p>A character vector, as the input protein sequence.
Use '<code>-</code>' to represent gaps in the sequence.</p></td>
    </tr>
    <tr>
      <th>propmat</th>
      <td><p>A matrix containing the properties for the amino acids.
Each row represent one amino acid type, each column represents
one property. Note that the one-letter row names must be provided
for we need them to seek the properties for each AA type.</p></td>
    </tr>
    <tr>
      <th>pc</th>
      <td><p>Integer. Use the first pc principal components as the scales.
Must be no greater than the number of AA properties provided.</p></td>
    </tr>
    <tr>
      <th>lag</th>
      <td><p>The lag parameter. Must be less than the amino acids.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>Logical. Should we auto-scale the property matrix
(<code>propmat</code>) before PCA? Default is <code>TRUE</code>.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>Logical. Whether to print the standard deviation,
proportion of variance and the cumulative proportion of
the selected principal components or not. Default is <code>TRUE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A length <code>lag * p^2</code> named vector,
<code>p</code> is the number of scales (principal components) selected.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>See <code><a href='extractProtFPGap.html'>extractProtFPGap</a></code> for amino acid property based
scales descriptors (protein fingerprint) with gap support.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'># amino acid sequence with gaps</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/align.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))$<span class='no'>`IXI_235`</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>AAindex</span>)
<span class='no'>AAidxmat</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span>(<span class='fu'><a href='https://rdrr.io/r/stats/na.fail.html'>na.omit</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>as.matrix</a></span>(<span class='no'>AAindex</span>[, <span class='fl'>7</span>:<span class='fl'>26</span>])))
<span class='no'>scales</span> <span class='kw'>&lt;-</span> <span class='fu'>extractScalesGap</span>(<span class='no'>x</span>, <span class='kw'>propmat</span> <span class='kw'>=</span> <span class='no'>AAidxmat</span>, <span class='kw'>pc</span> <span class='kw'>=</span> <span class='fl'>5</span>, <span class='kw'>lag</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>)</div><div class='output co'>#&gt; Summary of the first 5 principal components: 
#&gt;                             PC1      PC2      PC3      PC4     PC5
#&gt; Standard deviation     12.38381 10.73268 7.742507 6.802462 5.22316
#&gt; Proportion of Variance  0.28881  0.21693 0.112890 0.087140 0.05138
#&gt; Cumulative Proportion   0.28881  0.50574 0.618640 0.705780 0.75716</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


