<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ protr</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="Function reference" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/road2stat/protr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-protein-peptide-sequence-descriptors" class="hasAnchor"><a href="#section-protein-peptide-sequence-descriptors" class="anchor"></a>Protein/Peptide Sequence Descriptors</h2>
          <p class="section-desc"><p>Functions for calculating protein/peptide sequence descirptors.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="extractAAC.html">extractAAC()</a></code> </p>
        </td>
        <td><p>Amino Acid Composition Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractDC.html">extractDC()</a></code> </p>
        </td>
        <td><p>Dipeptide Composition Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractTC.html">extractTC()</a></code> </p>
        </td>
        <td><p>Tripeptide Composition Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractMoreauBroto.html">extractMoreauBroto()</a></code> </p>
        </td>
        <td><p>Normalized Moreau-Broto Autocorrelation Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractMoran.html">extractMoran()</a></code> </p>
        </td>
        <td><p>Moran Autocorrelation Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractGeary.html">extractGeary()</a></code> </p>
        </td>
        <td><p>Geary Autocorrelation Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDC.html">extractCTDC()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Composition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDCClass.html">extractCTDCClass()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Composition (with customized amino acid
classification support)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDDClass.html">extractCTDDClass()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Distribution (with customized amino acid
classification support)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDTClass.html">extractCTDTClass()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Transition (with customized amino acid
classification support)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDT.html">extractCTDT()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Transition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTDD.html">extractCTDD()</a></code> </p>
        </td>
        <td><p>CTD Descriptors - Distribution</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTriad.html">extractCTriad()</a></code> </p>
        </td>
        <td><p>Conjoint Triad Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractCTriadClass.html">extractCTriadClass()</a></code> </p>
        </td>
        <td><p>Conjoint Triad Descriptor (with customized amino acid
classification support)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractSOCN.html">extractSOCN()</a></code> </p>
        </td>
        <td><p>Sequence-Order-Coupling Numbers</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractQSO.html">extractQSO()</a></code> </p>
        </td>
        <td><p>Quasi-Sequence-Order (QSO) Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractPAAC.html">extractPAAC()</a></code> </p>
        </td>
        <td><p>Pseudo Amino Acid Composition (PseAAC) Descriptor</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractAPAAC.html">extractAPAAC()</a></code> </p>
        </td>
        <td><p>Amphiphilic Pseudo Amino Acid Composition (APseAAC) Descriptor</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-profile-based-protein-peptide-sequence-descriptors" class="hasAnchor"><a href="#section-profile-based-protein-peptide-sequence-descriptors" class="anchor"></a>Profile-Based Protein/Peptide Sequence Descriptors</h2>
          <p class="section-desc"><p>Functions for calculating the profile-based (PSSM) descirptors.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="extractPSSM.html">extractPSSM()</a></code> </p>
        </td>
        <td><p>Compute PSSM (Position-Specific Scoring Matrix) for given protein sequence</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractPSSMAcc.html">extractPSSMAcc()</a></code> </p>
        </td>
        <td><p>Profile-based protein representation derived by PSSM
(Position-Specific Scoring Matrix) and auto cross covariance</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractPSSMFeature.html">extractPSSMFeature()</a></code> </p>
        </td>
        <td><p>Profile-based protein representation derived by PSSM
(Position-Specific Scoring Matrix)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="acc.html">acc()</a></code> </p>
        </td>
        <td><p>Auto Cross Covariance (ACC) for Generating Scales-Based Descriptors
of the Same Length</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-proteochemometric-modeling-pcm-descriptors" class="hasAnchor"><a href="#section-proteochemometric-modeling-pcm-descriptors" class="anchor"></a>Proteochemometric Modeling (PCM) Descriptors</h2>
          <p class="section-desc"><p>Functions for calculating proteochemometric modeling (PCM) descriptors.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="extractScales.html">extractScales()</a></code> </p>
        </td>
        <td><p>Scales-Based Descriptors derived by Principal Components Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractScalesGap.html">extractScalesGap()</a></code> </p>
        </td>
        <td><p>Scales-Based Descriptors derived by Principal Components Analysis
(with Gap Support)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractProtFP.html">extractProtFP()</a></code> </p>
        </td>
        <td><p>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractProtFPGap.html">extractProtFPGap()</a></code> </p>
        </td>
        <td><p>Amino Acid Properties Based Scales Descriptors (Protein Fingerprint)
with Gap Support</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractDescScales.html">extractDescScales()</a></code> </p>
        </td>
        <td><p>Scales-Based Descriptors with 20+ classes of Molecular Descriptors</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractFAScales.html">extractFAScales()</a></code> </p>
        </td>
        <td><p>Scales-Based Descriptors derived by Factor Analysis</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractMDSScales.html">extractMDSScales()</a></code> </p>
        </td>
        <td><p>Scales-Based Descriptors derived by Multidimensional Scaling</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="extractBLOSUM.html">extractBLOSUM()</a></code> </p>
        </td>
        <td><p>BLOSUM and PAM Matrix-Derived Descriptors</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-similarity-measures-between-proteins" class="hasAnchor"><a href="#section-similarity-measures-between-proteins" class="anchor"></a>Similarity Measures between Proteins</h2>
          <p class="section-desc"><p>Functions for calculating protein sequence alignment based similarity measures and GO-based semantic similarity measures.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="twoSeqSim.html">twoSeqSim()</a></code> </p>
        </td>
        <td><p>Protein Sequence Alignment for Two Protein Sequences</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="parSeqSim.html">parSeqSim()</a></code> </p>
        </td>
        <td><p>Parallellized Protein Sequence Similarity Calculation based on
Sequence Alignment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="twoGOSim.html">twoGOSim()</a></code> </p>
        </td>
        <td><p>Protein Similarity Calculation based on Gene Ontology (GO) Similarity</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="parGOSim.html">parGOSim()</a></code> </p>
        </td>
        <td><p>Protein Similarity Calculation based on Gene Ontology
(GO) Similarity</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-helper-functions-for-pre-processing-protein-sequences" class="hasAnchor"><a href="#section-helper-functions-for-pre-processing-protein-sequences" class="anchor"></a>Helper Functions for Pre-Processing Protein Sequences</h2>
          <p class="section-desc"><p>Functions for pre-processing protein sequences.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="getUniProt.html">getUniProt()</a></code> </p>
        </td>
        <td><p>Retrieve Protein Sequences from UniProt by Protein ID</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readFASTA.html">readFASTA()</a></code> </p>
        </td>
        <td><p>Read Protein Sequences in FASTA Format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="readPDB.html">readPDB()</a></code> </p>
        </td>
        <td><p>Read Protein Sequences in PDB Format</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="protcheck.html">protcheck()</a></code> </p>
        </td>
        <td><p>Protein sequence amino acid type sanity check</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="protseg.html">protseg()</a></code> </p>
        </td>
        <td><p>Protein Sequence Segmentation/Partition</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="removeGaps.html">removeGaps()</a></code> </p>
        </td>
        <td><p>Remove or replace gaps from protein sequences.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="protr-package.html">protr-package</a></code> </p>
        </td>
        <td><p>Generating Various Numerical Representation Schemes for Protein Sequences</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-precomputed-molecular-descriptors-for-the-amino-acids" class="hasAnchor"><a href="#section-precomputed-molecular-descriptors-for-the-amino-acids" class="anchor"></a>Precomputed Molecular Descriptors for the 20 Amino Acids</h2>
          <p class="section-desc"><p>Precomputed molecular descriptors for the 20 amino acids.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="AAindex.html">AAindex</a></code> </p>
        </td>
        <td><p>AAindex Data of 544 Physicochemical and Biological Properties
for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAMetaInfo.html">AAMetaInfo</a></code> </p>
        </td>
        <td><p>Meta Information for the 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="OptAA3d.html">OptAA3d</a></code> </p>
        </td>
        <td><p>OptAA3d.sdf - 20 Amino Acids Optimized with MOE 2011.10 (Semiempirical AM1)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AA2DACOR.html">AA2DACOR</a></code> </p>
        </td>
        <td><p>2D Autocorrelations Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AA3DMoRSE.html">AA3DMoRSE</a></code> </p>
        </td>
        <td><p>3D-MoRSE Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAACF.html">AAACF</a></code> </p>
        </td>
        <td><p>Atom-Centred Fragments Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AABurden.html">AABurden</a></code> </p>
        </td>
        <td><p>Burden Eigenvalues Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAConn.html">AAConn</a></code> </p>
        </td>
        <td><p>Connectivity Indices Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAConst.html">AAConst</a></code> </p>
        </td>
        <td><p>Constitutional Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AACPSA.html">AACPSA</a></code> </p>
        </td>
        <td><p>CPSA Descriptors for 20 Amino Acids calculated by Discovery Studio</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AADescAll.html">AADescAll</a></code> </p>
        </td>
        <td><p>All 2D Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAEdgeAdj.html">AAEdgeAdj</a></code> </p>
        </td>
        <td><p>Edge Adjacency Indices Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAEigIdx.html">AAEigIdx</a></code> </p>
        </td>
        <td><p>Eigenvalue-Based Indices Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAFGC.html">AAFGC</a></code> </p>
        </td>
        <td><p>Functional Group Counts Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAGeom.html">AAGeom</a></code> </p>
        </td>
        <td><p>Geometrical Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAGETAWAY.html">AAGETAWAY</a></code> </p>
        </td>
        <td><p>GETAWAY Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAInfo.html">AAInfo</a></code> </p>
        </td>
        <td><p>Information Indices Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAMOE2D.html">AAMOE2D</a></code> </p>
        </td>
        <td><p>2D Descriptors for 20 Amino Acids calculated by MOE 2011.10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAMOE3D.html">AAMOE3D</a></code> </p>
        </td>
        <td><p>3D Descriptors for 20 Amino Acids calculated by MOE 2011.10</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAMolProp.html">AAMolProp</a></code> </p>
        </td>
        <td><p>Molecular Properties Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AARandic.html">AARandic</a></code> </p>
        </td>
        <td><p>Randic Molecular Profiles Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AARDF.html">AARDF</a></code> </p>
        </td>
        <td><p>RDF Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AATopo.html">AATopo</a></code> </p>
        </td>
        <td><p>Topological Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AATopoChg.html">AATopoChg</a></code> </p>
        </td>
        <td><p>Topological Charge Indices Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAWalk.html">AAWalk</a></code> </p>
        </td>
        <td><p>Walk and Path Counts Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAWHIM.html">AAWHIM</a></code> </p>
        </td>
        <td><p>WHIM Descriptors for 20 Amino Acids calculated by Dragon</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-blosum-and-pam-matrices-for-the-amino-acids" class="hasAnchor"><a href="#section-blosum-and-pam-matrices-for-the-amino-acids" class="anchor"></a>BLOSUM and PAM matrices for the 20 Amino Acids</h2>
          <p class="section-desc"><p>BLOSUM and PAM matrices for the 20 amino acids.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="AABLOSUM45.html">AABLOSUM45</a></code> </p>
        </td>
        <td><p>BLOSUM45 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AABLOSUM50.html">AABLOSUM50</a></code> </p>
        </td>
        <td><p>BLOSUM50 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AABLOSUM62.html">AABLOSUM62</a></code> </p>
        </td>
        <td><p>BLOSUM62 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AABLOSUM80.html">AABLOSUM80</a></code> </p>
        </td>
        <td><p>BLOSUM80 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AABLOSUM100.html">AABLOSUM100</a></code> </p>
        </td>
        <td><p>BLOSUM100 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAPAM30.html">AAPAM30</a></code> </p>
        </td>
        <td><p>PAM30 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAPAM40.html">AAPAM40</a></code> </p>
        </td>
        <td><p>PAM40 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAPAM70.html">AAPAM70</a></code> </p>
        </td>
        <td><p>PAM70 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAPAM120.html">AAPAM120</a></code> </p>
        </td>
        <td><p>PAM120 Matrix for 20 Amino Acids</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AAPAM250.html">AAPAM250</a></code> </p>
        </td>
        <td><p>PAM250 Matrix for 20 Amino Acids</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-protein-peptide-sequence-descriptors">Protein/Peptide Sequence Descriptors</a></li>
      <li><a href="#section-profile-based-protein-peptide-sequence-descriptors">Profile-Based Protein/Peptide Sequence Descriptors</a></li>
      <li><a href="#section-proteochemometric-modeling-pcm-descriptors">Proteochemometric Modeling (PCM) Descriptors</a></li>
      <li><a href="#section-similarity-measures-between-proteins">Similarity Measures between Proteins</a></li>
      <li><a href="#section-helper-functions-for-pre-processing-protein-sequences">Helper Functions for Pre-Processing Protein Sequences</a></li>
      <li><a href="#section-precomputed-molecular-descriptors-for-the-amino-acids">Precomputed Molecular Descriptors for the 20 Amino Acids</a></li>
      <li><a href="#section-blosum-and-pam-matrices-for-the-amino-acids">BLOSUM and PAM matrices for the 20 Amino Acids</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

