<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Parallellized Protein Sequence Similarity Calculation based on
Sequence Alignment (Disk-Based Version) — parSeqSimDisk • protr</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Parallellized Protein Sequence Similarity Calculation based on
Sequence Alignment (Disk-Based Version) — parSeqSimDisk" />
<meta property="og:description" content="This function implemented the parallellized version for calculating
protein sequence similarity based on sequence alignment.
This version caches the partial results in each batch to the
hard drive and merges the results together in the end, which
reduces the memory usage." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.6-2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/nanxstats/protr/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Parallellized Protein Sequence Similarity Calculation based on
Sequence Alignment (Disk-Based Version)</h1>
    <small class="dont-index">Source: <a href='https://github.com/nanxstats/protr/blob/master/R/par-01-parSeqSim.R'><code>R/par-01-parSeqSim.R</code></a></small>
    <div class="hidden name"><code>parSeqSimDisk.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function implemented the parallellized version for calculating
protein sequence similarity based on sequence alignment.
This version caches the partial results in each batch to the
hard drive and merges the results together in the end, which
reduces the memory usage.</p>
    </div>

    <pre class="usage"><span class='fu'>parSeqSimDisk</span>(
  <span class='no'>protlist</span>,
  <span class='kw'>cores</span> <span class='kw'>=</span> <span class='fl'>2</span>,
  <span class='kw'>batches</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>path</span> <span class='kw'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/tempfile.html'>tempdir</a></span>(),
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"local"</span>,
  <span class='kw'>submat</span> <span class='kw'>=</span> <span class='st'>"BLOSUM62"</span>,
  <span class='kw'>gap.opening</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='kw'>gap.extension</span> <span class='kw'>=</span> <span class='fl'>4</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>protlist</th>
      <td><p>A length <code>n</code> list containing <code>n</code> protein sequences,
each component of the list is a character string, storing one protein
sequence. Unknown sequences should be represented as <code>""</code>.</p></td>
    </tr>
    <tr>
      <th>cores</th>
      <td><p>Integer. The number of CPU cores to use for parallel execution,
default is <code>2</code>. Users can use the <code>detectCores()</code> function
in the <code>parallel</code> package to see how many cores they could use.</p></td>
    </tr>
    <tr>
      <th>batches</th>
      <td><p>Integer. How many batches should we split the pairwise
similarity computations into. This is useful when you have a large
number of protein sequences, enough number of CPU cores, but not
enough RAM to compute and hold all the pairwise similarities
in a single batch. Defaults to 1.</p></td>
    </tr>
    <tr>
      <th>path</th>
      <td><p>Directory for caching the results in each batch.
Defaults to the temporary directory.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Print the computation progress?</p></td>
    </tr>
    <tr>
      <th>type</th>
      <td><p>Type of alignment, default is <code>"local"</code>,
can be <code>"global"</code> or <code>"local"</code>,
where <code>"global"</code> represents Needleman-Wunsch global alignment;
<code>"local"</code> represents Smith-Waterman local alignment.</p></td>
    </tr>
    <tr>
      <th>submat</th>
      <td><p>Substitution matrix, default is <code>"BLOSUM62"</code>,
can be one of <code>"BLOSUM45"</code>, <code>"BLOSUM50"</code>, <code>"BLOSUM62"</code>,
<code>"BLOSUM80"</code>, <code>"BLOSUM100"</code>, <code>"PAM30"</code>,
<code>"PAM40"</code>, <code>"PAM70"</code>, <code>"PAM120"</code>, or <code>"PAM250"</code>.</p></td>
    </tr>
    <tr>
      <th>gap.opening</th>
      <td><p>The cost required to open a gap of any length
in the alignment. Defaults to 10.</p></td>
    </tr>
    <tr>
      <th>gap.extension</th>
      <td><p>The cost to extend the length of an existing
gap by 1. Defaults to 4.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>n</code> x <code>n</code> similarity matrix.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p>See <code><a href='parSeqSim.html'>parSeqSim</a></code> for the in-memory version.</p></div>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {

<span class='co'># Be careful when testing this since it involves parallelisation</span>
<span class='co'># and might produce unpredictable results in some environments</span>

<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"Biostrings"</span>)
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"foreach"</span>)
<span class='fu'><a href='https://rdrr.io/r/base/library.html'>library</a></span>(<span class='st'>"doParallel"</span>)

<span class='no'>s1</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/P00750.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s2</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/P08218.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s3</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/P10323.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s4</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/P20160.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='no'>s5</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='readFASTA.html'>readFASTA</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span>(<span class='st'>"protseq/Q9NZP8.fasta"</span>, <span class='kw'>package</span> <span class='kw'>=</span> <span class='st'>"protr"</span>))<span class='kw'>[[</span><span class='fl'>1</span>]]
<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span>(<span class='fl'>1010</span>)
<span class='no'>plist</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>as.list</a></span>(<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='no'>s1</span>, <span class='no'>s2</span>, <span class='no'>s3</span>, <span class='no'>s4</span>, <span class='no'>s5</span>)[<span class='fu'><a href='https://rdrr.io/r/base/sample.html'>sample</a></span>(<span class='fl'>1</span>:<span class='fl'>5</span>, <span class='fl'>100</span>, <span class='kw'>replace</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)])
<span class='no'>psimmat</span> <span class='kw'>&lt;-</span> <span class='fu'>parSeqSimDisk</span>(
  <span class='no'>plist</span>,
  <span class='kw'>cores</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>batches</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>type</span> <span class='kw'>=</span> <span class='st'>"local"</span>, <span class='kw'>submat</span> <span class='kw'>=</span> <span class='st'>"BLOSUM62"</span>
)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


