<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generating Various Numerical Representation Schemes for Protein Sequences — protr-package • protr</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cerulean/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Generating Various Numerical Representation Schemes for Protein Sequences — protr-package" />

<meta property="og:description" content="The protr package is a comprehensive toolkit for generating various
numerical representation schemes of protein sequence.
The descriptors are extensively utilized in bioinformatics and
chemogenomics research. The commonly used descriptors include
amino acid composition, autocorrelation, CTD, conjoint traid,
quasi-sequence order, pseudo amino acid composition, and
profile-based descriptors derived by Position-Specific Scoring Matrix (PSSM).
The descriptors for proteochemometric (PCM) modeling include the scales-based
descriptors derived by principal components analysis, factor analysis,
multidimensional scaling, amino acid properties (AAindex), 20+ classes
of 2D and 3D molecular descriptors (Topological, WHIM, VHSE, etc.),
and BLOSUM/PAM matrix-derived descriptors. The protr package also
integrates the function of parallelized similarity computation derived
by pairwise protein sequence alignment and Gene Ontology (GO) semantic
similarity measures." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-66394392-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-66394392-1');
</script>

  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">protr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/protr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/road2stat/protr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generating Various Numerical Representation Schemes for Protein Sequences</h1>
    <small class="dont-index">Source: <a href='https://github.com/road2stat/protr/blob/master/R/protr-package.R'><code>R/protr-package.R</code></a></small>
    <div class="hidden name"><code>protr-package.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>The protr package is a comprehensive toolkit for generating various
numerical representation schemes of protein sequence.
The descriptors are extensively utilized in bioinformatics and
chemogenomics research. The commonly used descriptors include
amino acid composition, autocorrelation, CTD, conjoint traid,
quasi-sequence order, pseudo amino acid composition, and
profile-based descriptors derived by Position-Specific Scoring Matrix (PSSM).
The descriptors for proteochemometric (PCM) modeling include the scales-based
descriptors derived by principal components analysis, factor analysis,
multidimensional scaling, amino acid properties (AAindex), 20+ classes
of 2D and 3D molecular descriptors (Topological, WHIM, VHSE, etc.),
and BLOSUM/PAM matrix-derived descriptors. The protr package also
integrates the function of parallelized similarity computation derived
by pairwise protein sequence alignment and Gene Ontology (GO) semantic
similarity measures.</p>
    
    </div>

        
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <table class='table'>
<tr><td>Package:</td><td>protr</td></tr>
<tr><td>Type:</td><td>Package</td></tr>
<tr><td>License:</td><td>BSD_3_clause</td></tr>
<tr><td></td><td>Package:</td></tr>
</table>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>The package vignette can be opened with <code><a href='../articles/protr.html'>vignette("protr")</a></code>.</p>
<p>The web server for this package, <code>ProtrWeb</code> can be accessed from:
<a href='http://protr.org'>http://protr.org</a>.</p>
<p>Bug reports and feature requests should be sent to
<a href='https://github.com/road2stat/protr/issues'>https://github.com/road2stat/protr/issues</a>.</p>
    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Xiao, N., Cao, D.-S., Zhu, M.-F., and Xu, Q.-S. (2015).
protr/ProtrWeb: R package and web server for generating various
numerical representation schemes of protein sequences.
<em>Bioinformatics</em> 31 (11), 1857--1859.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      
      <li><a href="#details">Details</a></li>

      <li><a href="#note">Note</a></li>

      <li><a href="#references">References</a></li>
          </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Nan Xiao, Qing-Song Xu, Dong-Sheng Cao.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

